test/initial_release/main.d

```diff
@@ -0,0 +1,172 @@
+	listenHTTP(settings, router);
+
+    import std.functional : toDelegate;
+
+    auto task = runTask(toDelegate(&runNeptune), gh_test);
+
+    runEventLoop();
+
+    task.join();
+
+    assert(!failed);
+}
+
+/*******************************************************************************
+
+    Runs the test
+
+    Params:
+        gh_test = github mockup server instance
+
+*******************************************************************************/
+
+void runNeptune ( RestAPI gh_test )
+{
+    import test.initial_release.shellHelper;
+
+    import vibe.core.core;
+    import vibe.core.args;
+
+    import std.stdio;
+    import std.process;
+    import std.format;
+
+    import core.time;
+
+    enum GitRepo = "tester/sandbox";
+    enum GitPath = "github.com:" ~ GitRepo;
+
+    scope(failure)
+        failed = true;
+
+    scope(exit)
+        exitEventLoop();
+
+    auto tmp = readRequiredOption!string("tmp", "Temporary directory");
+    auto bin = readRequiredOption!string("bin", "Binary directory");
+    auto data = __FILE_FULL_PATH__[0 .. $-"main.d".length] ~ "data";
+
+    auto git = tmp ~ "/" ~ GitPath;
+
+    tmp.cmd("rm -rf " ~ GitPath);
+    tmp.cmd("mkdir -p " ~ GitPath);
+    git.cmd("git init");
+    git.cmd("git config neptune.upstream " ~ GitRepo);
+    git.cmd("git config neptune.oauthtoken 0000");
+    git.cmd("git config neptune.mail-recipient dummy@notexisting.example");
+    git.cmd("git remote add origin " ~ git);
+
+    git.cmd("git checkout -B v1.x.x");
+
+    git.cmd(format("cp -r %s/relnotes ./", data));
+
+    git.cmd("git add relnotes");
+    git.cmd(`git commit -m "Add rel notes"`);
+
+    auto neptune = pipeProcess([format("%s/neptune-release", bin),
+                               "--assume-yes=true",
+                               "--no-send-mail",
+                               "--verbose",
+                               "--save-mail",
+                               "--base-url=http://127.0.0.1:8080"],
+                               Redirect.all, null, Config.none, git);
+
+    bool killed = false;
+
+    void timeout ( )
+    {
+        kill(neptune.pid);
+        killed = true;
+    }
+
+    // fallback timeout timer
+    setTimer(10.seconds, &timeout);
+
+
+    string ret, line;
+
+    import vibe.stream.stdio;
+    import vibe.stream.operations;
+
+    auto _stdout = new StdFileStream(true, false);
+    auto _stderr = new StdFileStream(true, false);
+
+    _stdout.setup(neptune.stdout);
+    _stderr.setup(neptune.stderr);
+
+    auto stderr = _stderr.readAllUTF8();
+    auto stdout = _stdout.readAllUTF8();
+
+    scope(failure)
+    {
+        writefln("An assert failed. Neptune output was:\n------\n%s\n-----\n%s",
+                 stderr, stdout);
+    }
+
+    assert(!killed);
+
+    auto w = neptune.pid.tryWait();
+
+    assert(w.terminated);
+    assert(w.status == 0);
+
+    import std.stdio;
+    import std.string : strip;
+    import std.algorithm : startsWith;
+
+    sleep(200.msecs);
```

Comment:
The 10s times before looks OK as a safety net, but a 200ms timer in a test screams flakeyness in my eyes. Are you sure this is a good idea?